plugins/experimental/traffic_dump/traffic_dump.cc

```diff
@@ -315,12 +332,175 @@ session_aio_handler(TSCont contp, TSEvent event, void *edata)
     return TS_SUCCESS;
   }
   default:
-    TSDebug(PLUGIN_NAME, "session_aio_handler(): unhandled events %d", event);
+    TSDebug(PLUGIN_NAME, "%s: unhandled events %d", __func__, event);
     return TS_ERROR;
   }
   return TS_SUCCESS;
 }
 
+std::string
+request_body_get(TSHttpTxn txnp)
+{
+  TSDebug(PLUGIN_NAME, "getting the body");
+  TSIOBufferReader post_buffer_reader = TSHttpTxnPostBufferReaderGet(txnp);
+  int64_t read_avail                  = TSIOBufferReaderAvail(post_buffer_reader);
+  if (read_avail == 0) {
+    TSIOBufferReaderFree(post_buffer_reader);
+    return "";
+  }
+
+  // std::string has no reserving allocator. So create an explicitly empty
+  // string then reserve the bytes for later population.
+  std::string body_string;
+  body_string.reserve(read_avail + 1); // '+ 1' for null termination.
+
```

Comment:
Possibly
```
std::string body_string;
body_string.resize(read_avail+1);
TSIOBufferReaderCopy(post_buffer_reader, body_string.data(), read_avail+1);
```
rather than doing the loop.